src/MicrosoftAuthFlow.js

```diff
@@ -113,39 +113,39 @@ class MicrosoftAuthFlow {
 
   async getXboxToken (relyingParty = this.options.relyingParty || Endpoints.XboxRelyingParty) {
     const options = { ...this.options, relyingParty }
-    if (await this.xbl.verifyTokens(relyingParty)) {
+
+    const { xstsToken, userToken, deviceToken, titleToken } = await this.xbl.getCachedTokens(relyingParty)
+
+    if (xstsToken.valid) {
       debug('[xbl] Using existing XSTS token')
-      const { data } = await this.xbl.getCachedXstsToken(relyingParty)
-      return data
-    } else if (options.password) {
+      return xstsToken.data
+    }
+
+    if (options.password) {
       debug('[xbl] password is present, trying to authenticate using xboxreplay/xboxlive-auth')
       const xsts = await this.xbl.doReplayAuth(this.username, options.password, options)
       return xsts
-    } else {
-      debug('[xbl] Need to obtain tokens')
-      return await retry(async () => {
-        const msaToken = await this.getMsaToken()
+    }
 
-        if (options.flow === 'sisu') {
-          debug(`[xbl] Sisu flow selected, trying to authenticate with authTitle ID ${options.authTitle}`)
-          const deviceToken = await this.xbl.getDeviceToken(options)
-          const sisu = await this.xbl.doSisuAuth(msaToken, deviceToken, options)
-          return sisu
-        }
+    debug('[xbl] Need to obtain tokens')
 
-        const userToken = await this.xbl.getUserToken(msaToken, options.flow === 'msal')
+    return await retry(async () => {
+      const msaToken = await this.getMsaToken()
 
-        if (this.doTitleAuth) {
-          const deviceToken = await this.xbl.getDeviceToken(options)
-          const titleToken = await this.xbl.getTitleToken(msaToken, deviceToken)
-          const xsts = await this.xbl.getXSTSToken({ userToken, deviceToken, titleToken }, options)
-          return xsts
-        } else {
-          const xsts = await this.xbl.getXSTSToken({ userToken }, options)
-          return xsts
-        }
-      }, () => { this.msa.forceRefresh = true }, 2)
-    }
+      if (options.flow === 'sisu' && (!userToken.valid || !deviceToken.valid || !titleToken.valid)) {
```

Comment:
Can you please explain this:
`      if (options.flow === 'sisu' && (!userToken.valid || !deviceToken.valid || !titleToken.valid)) {`?

Reply:
Sure, so with `sisu` we generate a devcieToken as usual and in return `sisu` will retrieve userToken & titleToken from the same call. So here we're checking if any of those tokens are invalid and then using `sisu` flow to refresh.

Reply:
ok, might be helpful to add a comment here explaining that (that if =='sisu', you don't expect a refresh to be triggered below as would happen with the nullish `??` handling).